src/widget/taskBar.js

```diff
@@ -620,7 +639,11 @@ let TileableItem = GObject.registerClass(
 
         // Update the title and crop it if it's too long
         updateTitle() {
-            this.title.text = this.tileable.title;
+            if(this.style == 'full') {
+                this.title.text = this.tileable.title;
```

Comment:
In my idea the `full` mode would display both title and app name 
```
this.title.text = `${this.tileable.title} - ${this.app.get_name()}`;
```
Therefore it's should most often reach the maximum width and being less distractive